lib/tasks/github.rake

```diff
@@ -0,0 +1,56 @@
+require 'literate_randomizer'
+require 'octokit'
+
+namespace :github do
+  desc "Sync database authors with those specificed by the Github issues as per Feature-001"
+  task sync: :environment do
+    # Get issues from github, which represent individual 
+    # authors and their associated biography
+    client = Octokit::Client.new
+    client.auto_paginate = true
+    issues = client.issues 'scasagrande/bookstore-authors'
+    
+    # Hash github issue ID and issue data
+    authors_gh_h = Hash.new
+    issues.each do |issue|
+      authors_gh_h[issue[:number]] = issue
+    end
+
+    # Retrieve from the DB the list of authors that have matching names
+    # to our github list of authors.
+    puts "Updating existing authors that have been modified"
```

Comment:
I understand that this block isn't very performant. We're gathering all matching authors from the DB, looking through each one, then performing a DB update one author at a time.

Aside from the update call potentially being batched, the other performance would be to only retrieve the list of authors from the DB that are out of date. However, I wasn't sure of how to go about conditional queries on tuples of data (`id` paired with a check on `updated_at`) in some sort of standard rails way. A custom SQL query could be done, but I figured for something that will generally be run infrequently that it would be better to get something working, then figure out if we need to optimize it a little more.

Reply:
The query _could_ be improved in a rails-standard way to do a bunch of `OR`'d conditions in order to just get the records that are actually out of date, but the resulting code is kinda messy (I tried to write it here as an example) and the readability penalty probably isn't worth it unless / until this becomes an actual performance bottleneck. 

Rails does have an `in_batches` method you can use to loop over your `authors_db_to_keep` though, which will only load a subset of records into memory at any given time and will really help performance by putting an upper bound on the number of records being read from the DB or being stored in RAM at any given time, without adding much code complexity.